### PR TITLE
Bug 1827686 - Add telemetry for user updating Default Search Engine

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1827686

### CODE REVIEW

**Comment by rocketsroger** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us know what user prefers as default search engine.

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This will allow us to know more about user's engagement with default search engine settings.
 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
User updated default search engine | Category 2 - Interaction data | https://github.com/mozilla-mobile/firefox-android/pull/1674

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * Until release version 127 with the option to renew at that point.

8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Glam. 
11. Where do you intend to share the results of your analysis?
  * Only on Glean, Glam, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

**Comment by eliserichards** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us know what user prefers as default search engine.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
>     * This will allow us to know more about user's engagement with default search engine settings.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * There are no other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> User updated default search engine 	Category 2 - Interaction data 	#1674
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix%E2%80%A8)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Until release version 127 with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Glam.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean, Glam, and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

* Yes, through `metrics.yaml` and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? (Note, for data collection not needed for security purposes, Mozilla provides such a control mechanism) Provide details as to the control mechanism available.

* Yes, through the "Send Usage Data" preference in the application settings.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

* This will be monitored until version 127 with the option to remove or renew at that point.

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

* Category 2 - interaction data.

5) Is the data collection request for default-on or default-off?

* Default on.

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

* No. 

7) Is the data collection covered by the existing Firefox privacy notice? **If unsure: escalate to legal if:**

* Yes.

8) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

* No.

## Result

data-review+

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt [hunk 1]
```diff
@@ -156,6 +157,14 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
         )
 
         context.components.useCases.searchUseCases.selectSearchEngine(engine)
+
+        val searchEngineName = when (engine.type) {
+            SearchEngine.Type.BUNDLED, SearchEngine.Type.BUNDLED_ADDITIONAL -> engine.name
+            SearchEngine.Type.CUSTOM -> "Custom"
+            SearchEngine.Type.APPLICATION -> "Application"
+        }
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt:165]: Is this intentionally deviating from this same metric in `SearchDialogController`?

https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt#241-251

Comment (rocketsroger) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt:165]: Forgot we did that as well.  That's a good idea, I'll make that available as a extension.  Thanks!

Comment (rocketsroger) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt]: @MozillaNoah Please let me know if this is what you mean.  I've moved the code you linked to the extension so we can use it for this change as well.  Thanks

**Review by MozillaNoah — APPROVED** [review]
:shipit:

File: fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt]: The refactor looks great! :shipit:

**Pull request merged**